invenio_app_rdm/communities_ui/views/communities.py

```diff
@@ -126,3 +129,34 @@ def community_static_page(pid_value, community, community_ui, **kwargs):
         community=community_ui,
         permissions=permissions,
     )
+
+
+@pass_community(serialize=True)
+def community_collection(
+    community, community_ui, pid_value, tree_slug, collection_slug
+):
+    """Render a community collection page."""
+    collections_service = current_rdm_records.collections_service
+    collection = None
+    try:
+        collection = collections_service.read_slug(
+            g.identity,
+            community_id=community.id,
+            slug=collection_slug,
+            tree_slug=tree_slug,
+        )
+    except Exception:
+        abort(404)
```

Comment:
Will this be improved in the following prs?

Reply:
Yes, alongside adding specific errors to the collections backend.